src/dra/8D3E8.c

```diff
@@ -764,7 +764,194 @@ void func_8012EF2C(void) {
 #endif
 }
 
-INCLUDE_ASM("dra/nonmatchings/8D3E8", func_8012F178);
+void func_8012F178(Primitive* prim, s32 count, bool finishUp) {
+    // Someone got the brilliant idea to use the scratchpad for every
+    // single local variable in this function. So this isn't really a
+    // struct, but it describes the layout of that scratchpad. It's not
+    // used anywhere else, so we just define it inside the function.
+    typedef struct {
+        s32 unused;
+        s32 i;
+        s32 x;
+        s32 y;
+        s32 velX;
+        s32 velY;
+        s32 i_limit;
+        Primitive* prim;
+        helper_8012F178* helper;
+        byte pad[15];
+        u8 colors[2];
+    } scratchpad_8012F178;
+    scratchpad_8012F178* scratchpad;
+
+#ifdef VERSION_PSP
+#define LEFTVEL -0x13
+#define TPAGE 0x4118
+#define V_OFFSET 0
+#else
+#define LEFTVEL -0x14
+#define TPAGE 0x118
+#define V_OFFSET 0xC0
+#endif
+#ifdef VERSION_PC
+    scratchpad_8012F178 _s;
+    scratchpad = &_s;
+#else
+    scratchpad = (scratchpad_8012F178*)SPAD(0);
+#endif
+    scratchpad->prim = prim;
```

Comment:
scratchpad is used so much in this function I kind of wonder if it should have a minimal name like
```
            s->prim->u0 = s->prim->u2 = 0;
            s->prim->u1 = s->prim->u3 = 0x3F;
            s->prim->v0 = s->prim->v1 = V_OFFSET;
            s->prim->v2 = s->prim->v3 = V_OFFSET + 0x3F;
            s->prim->tpage = TPAGE;
            s->prim->priority = PLAYER.zPriority - 6;
            s->prim->drawMode = 0x13F;
            s->helper->state++;
```
Should get approval from @Xeeynamo before changing though probably

Reply:
Yeah I think `s` or `sp` is enough. I _personally_ prefer shorter names for local variables as they are very targeted and self-contained with the function.
Also a good blog post that describes what to do in these kind of situations: https://www.lysator.liu.se/c/pikestyle.html

Reply:
I'll go with `s` since `sp` is used as the actual scratchpad for PC in `common.h`